test/lib/listeners/ProgressEventListener.test.ts

```diff
@@ -90,6 +90,42 @@ describe('ProgressEventListener implementations', () => {
 			expect(endEvent.data).toContain('done');
 		});
 
+		it('Properly aggregates percentages across multiple Cores', async () => {
+			// ==== TEST SETUP ====
+			// Instantiate a second Core and assign the standard functional stubs to both instances.
+			const secondCore: CodeAnalyzer = new CodeAnalyzer(CodeAnalyzerConfig.withDefaults());
+			const factory1 = new StubEnginePluginsFactory_withFunctionalStubEngine();
+			for (const enginePlugin of factory1.create()) {
+				await codeAnalyzer.addEnginePlugin(enginePlugin);
+			}
+			const factory2 = new StubEnginePluginsFactory_withFunctionalStubEngine();
+			for (const enginePlugin of factory2.create()) {
+				await secondCore.addEnginePlugin(enginePlugin);
+			}
+
+			// We don't want automated ticking to mess with the messages, so just turn it off for now.
+			spinner = new RuleSelectionProgressSpinner(spyDisplay, -1);
+
+			// ==== TESTED BEHAVIOR ====
+			// Start listening to the spinners, select some rules in each one, then stop listening.
+			spinner.listen(codeAnalyzer, secondCore);
+			await codeAnalyzer.selectRules(['all']);
+			await secondCore.selectRules(['all']);
```

Comment:
This is a pretty simple scenario, but it hits the functionality we want. I could have gone nuts with interleaving timing tests or whatever, but that felt like putting a hat on a hat.